thrust/testing/shuffle.cu

```diff
@@ -676,4 +818,15 @@ void TestShuffleEvenDistribution()
     }
   }
 }
+template <typename Vector>
+void TestShuffleEvenDistribution()
+{
+  TestShuffleEvenDistributionBase<thrust_shuffle, Vector>();
+}
+template <typename Vector>
+void TestShuffleEvenDistributionIterator()
+{
+  TestShuffleEvenDistributionBase<iterator_shuffle, Vector>();
+}
+
```

Comment:
Minor: Let's add an additional test that combines the `shuffle_iterator` as the index iterator of a `permutation_iterator`. I suspect that this will be a common usecase and explicitly testing these together would be worthwhile.